app/src/main/java/com/crossbowffs/nekosms/app/FilterEditorFragment.java

```diff
@@ -133,4 +149,46 @@ public void onItemSelected(AdapterView<?> parent, View view, int position, long
         caseMap.put(CaseSensitivity.SENSITIVE, resources.getString(R.string.filter_case_sensitive));
         return caseMap;
     }
+
+    private void testPattern() {
+        String text = mTestText.getText().toString();
+        boolean matches = false;
+
+        if (mPatternData.hasData()) {
```

Comment:
This should be refactored. Make validation occur on all filter modes, and make it return valid if the pattern is not a regex. This is what `validatePatternString` already does. Also, there are no invalid wildcard patterns.

Reply:
I will fix it.